src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,655 @@ public ByteArraySet createSmallSet() {
         return set;
     }
 
+    public List<ByteArray> getRandomByteArrays(int count, long seed) {
+        System.out.println("Generating random " + count + " Byte Array(s) with " + seed + " seed");
+        Random randomSeed = new Random(seed);
+        List<ByteArray> vals = Stream.generate(() -> RandomStringUtils.random(RANDOM_STRING_SIZE, 0, 0, true, true, null, randomSeed))
+                .limit(count)
+                .collect(Collectors.toSet())
+                .stream()
+                .map(str -> new ByteArray(str.getBytes()))
+                .collect(Collectors.toList());
+        assertEquals(count, vals.size());
+        return vals;
+    }
+
+    public List<ByteArray> getRandomByteArrays(int count) {
+        return getRandomByteArrays(count, RANDOM_SEED);
+    }
+
+    public void testAdd(int size) {
+        System.out.println("Testing add method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        List<ByteArray> insertedVals = new ArrayList<ByteArray>();
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+            assertTrue(set.contains(val));
+            insertedVals.add(val);
+            assertEquals(insertedVals.size(), set.size());
+        }
+
+        for (ByteArray val: vals) {
+            assertFalse(set.add(val));
+            assertTrue(set.contains(val));
+        }
+        assertEquals(size, set.size());
+
+        for (ByteArray val: vals) {
+            assertTrue(set.contains(val));
+        }
+    }
+
+    @Test
+    public void emptyAdd() {
+        testAdd(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallAdd() {
+        testAdd(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallAdd() {
+        testAdd(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularAdd() {
+        testAdd(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigAdd() {
+        testAdd(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigAdd() {
+        testAdd(SIZE_REALLY_BIG);
+    }
+
+    public long getDifferentRandomSeed() {
```

Comment:
Why is this here? This function breaks the reason for using a global random seed for reproducibility. `RANDOM_SEED` should be used for every call to `getRandomByteArrays()` that way a new `Random` object is created with the same seed value for the whole test package

Reply:
It is used for one test case where we need to have a different set of values created.